File: Backend/src/main/java/net/ethlny/discordhetic/discord_backend_hetic/payload/SignupRequest.java
package net.ethlny.discordhetic.discord_backend_hetic.payload;

import java.util.Set;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class SignupRequest {
    private String username;
    private String email;
    private String password;
    // Optionnel : liste des rôles à attribuer
    private Set<String> roles;
}

File: Frontend/Models/SignUpRequest.cs
namespace DiscordHeticWpf.Models
{
    public class SignUpRequest
    {
        public string username { get; set; }
        public string email { get; set; }
        public string password { get; set; }
        // If empty, the backend defaults to "USER"
        public string[] roles { get; set; }
    }
}
